.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+  SOLR_VERSION: 6.6.5
+
+jobs:
+
+  build-static:
+    name: Build static files
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run webpack build
+      run: npm run build:prod
+    - name: Upload built files
+      uses: actions/upload-artifact@v2
+      with:
+        name: bundles
+        path: bundles/
+    - name: Upload stats file
+      uses: actions/upload-artifact@v2
+      with:
+        name: webpack-stats.json
+        path: webpack-stats.json
+
+  ts-unit:
+    name: TypeScript unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run Jest
+      run: npx jest --collectCoverage
+    - name: Upload test coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        flags: typescript
+
+  py-unit:
+    name: Python unit tests
+    runs-on: ubuntu-latest
+    needs: build-static
+    strategy:
+      matrix:
+        python: [3.5, 3.6, 3.7]
+    services:
+      db:
+        image: mariadb
+        env:
+          MYSQL_DATABASE: sandco
+          MYSQL_ROOT_PASSWORD: sandco
+        ports:
+        - 3306:3306
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Cache pip
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ matrix.python }}-${{ hashFiles('requirements.txt') }}
+        restore-keys: |
+          pip-${{ matrix.python }}-${{ hashFiles('requirements.txt') }}
+          pip-${{ matrix.python }}
+          pip-
+    - name: Install dependencies
+      run: |
+        pip install wheel
+        pip install -r requirements.txt
+        pip install -r dev-requirements.txt
+    - name: Install and configure Solr
+      run: bash ci/config_solr.sh
+    - name: Download webpack bundle
+      uses: actions/download-artifact@v2
+      with:
+        name: bundles
+        path: bundles/
+    - name: Download webpack stats
```

Comment:
What are the webpack stats for? 

Larger question — where/how do you document github actions workflows? There should be comments or descriptions somewhere and I don't see any, but I don't know what the convention is for it either.

Reply:
it's kind of a misleading name; the stats file is useful to us because it includes full paths to each of the files that webpack creates, which is what `django-webpack-loader` uses to find and load the files. so it functions more like a manifest - the "stats" aren't used.

i'm not sure what the convention is - thus far I haven't found a workflow file with any documentation directly on it. there's documentation for published actions, but that usually takes the form of a README in the repo where the action resides. perhaps we could treat it like travis and add some minimal comments in the YAML.